routes/likeProductReviews.js

```diff
@@ -8,7 +8,30 @@ const challenges = require('../data/datacache').challenges
 const db = require('../data/mongodb')
 const insecurity = require('../lib/insecurity')
 
-
+module.exports = function productReviews () {
+  return (req, res, next) => {
+    const id = req.body.id
+    const user = insecurity.authenticatedUsers.from(req)
+    db.reviews.findOne({ _id: id }).then(review => {
+      var likedBy = review.likedBy
+      if (!likedBy.includes(user.data.email)) {
+        db.reviews.update(
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; NoSQL Injection

Unsanitized input from `the HTTP request body` flows into `update`, where it is used in an NoSQL query. This may result in an NoSQL Injection vulnerability.

<h6>
Line 18 | <a href="https://cwe.mitre.org/data/definitions/943.html">CWE-943</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 804</a> | <a href="https://learn.snyk.io/lesson/nosql-injection-attack/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 10 steps</b></summary>

`Step 1 - 4` https://github.com/mnathsnyk/juice-shop/blob/963a64061d299efe758780dbc7781ed29dc1e9dd/routes/likeProductReviews.js#L13

`Step 5 - 6` [routes/likeProductReviews.js#L15](https://github.com/mnathsnyk/juice-shop/blob/963a64061d299efe758780dbc7781ed29dc1e9dd/routes/likeProductReviews.js#L15)

`Step 7 - 9` [routes/likeProductReviews.js#L19](https://github.com/mnathsnyk/juice-shop/blob/963a64061d299efe758780dbc7781ed29dc1e9dd/routes/likeProductReviews.js#L19)

`Step 10` https://github.com/mnathsnyk/juice-shop/blob/963a64061d299efe758780dbc7781ed29dc1e9dd/routes/likeProductReviews.js#L18

</details>